src/main/java/com/todo/backend/ToDo.java

```diff
@@ -0,0 +1,126 @@
+package com.todo.backend;
+
+import java.time.Instant;
+
+public class ToDo {
+    // #region ################################ PROPERTIES
+    /**
+     * Priority of ToDo items. Mandatory.
+     */
+    enum Priority {
+        HIGH,
+        MEDIUM,
+        LOW
+    }
+
+    /**
+     * Auto-incremental ID. Cannot modify it, but delete the whole object.
+     */
+    final private int id;
+
+    /**
+     * ToDo List item's message. The main body.
+     */
+    private String text = null;
+
+    /**
+     * Due date if any. It serves not purpose other than 'importance' colors.
+     */
+    private Instant due_date = null;
+
+    /**
+     * One of "LOW", "MEDIUM" or "HIGH". Required for filtering items.
+     */
+    private Priority priority = Priority.LOW;
+
+    /**
+     * Whether or not this item is "done". Could be replaced by 'done_date'.
+     */
+    private boolean done = false;
+
+    /**
+     * Date when this item was marked as done. It re-assings only when undone ->
+     * done.
+     */
+    private Instant done_date = null;
+
+    /**
+     * Date when the item was originally created. For performance-checking purposes.
+     */
+    final private Instant creation_date;
+
+    /**
+     * Keeps track of ID's auto increment so it always is unique.
+     */
+    static protected int AUTO_INCREMENT = 0;
+    // #endregion
+
+    // #region ################################ CONSTRUCTOR
+    public ToDo(String text, Priority priority, Instant due_date) {
+        this.id = AUTO_INCREMENT;
+        this.text = text;
+        this.priority = priority;
+        this.due_date = due_date;
+        this.creation_date = Instant.now();
+        AUTO_INCREMENT++;
+    }
+    // #endregion
+
+    // #region ################################ GETTERS
+    public int getId() {
+        return id;
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public Priority getPriority() {
+        return priority;
+    }
+
+    public Instant getDue_date() {
+        return due_date;
+    }
+
+    public Instant getDone_date() {
+        return done_date;
+    }
+
+    public Instant getCreation_date() {
+        return creation_date;
+    }
+
+    public boolean getDone() {
+        return done;
+    }
+    // #endregion
+
+    // #region ################################ SETTERS
+    public void setText(String text) {
+        if (text == null)
+            throw new Error("Text field cannot be null");
+
+        this.text = text;
+    }
+
+    public void setPriority(Priority priority) {
+        if (priority == null)
```

Comment:
this kind of logic could be living in a validator class before doing anything with the values at the POJO